fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -339,12 +323,18 @@ def zero_grad(self, set_to_none: bool = False) -> None:
                 See :meth:`torch.optim.Optimizer.zero_grad` for details.
         """
 
+        if self.auto_refresh_trainable:
```

Comment:
this is orthogonal but a bugfix: if the model trainability changed, and people zero_grad, then some of the gradients could skip being set to zero, which would break the subsequent grad norm computation..